### ISSUE TITLE
[CI] Prevent duplicated target file paths in data files.

### ISSUE THREAD

**Issue by dekiel** [kyma-project/test-infra#6694]
**Description**

A `to` directive defines target file to store rendered prowjob definitions. If we have the same file defined in `to` directive more than one it's causing a race condition when rendering a prowjob definitions. A target file is getting different content, depending which duplicated `to` directive is processed last. Different file content cause random failures of pre-main-test-infra-validate-generated-files prowjob.

```
- to: ../../prow/jobs/kyma/kyma-gke-external-registry.yaml
        localSets:
          jobConfig_default:
```

**Expected result**

Rendertemplate detect duplicated `to` directive values and fail pre-main-test-infra-validate-generated-files, printing location of duplicates.

**Actual result**

Rendertemplate detect only different file content. It's possible to have different file content on two PRs with the same content od data file.

**Comment by dekiel** [comment]
@KacperMalachowski lets sync on that issue. I've learned that some time ago we got an contribution to that tool which introduced a FromTo struct which in fact bring a feature of multiple data files writing to the same target file. With that, the description in this issue seams wrong. If I remember correctly there might be an issue with order in which prowjobs are put in to shared target file, but I don't remember details now.
